lib/core/src/display/rendering/dom_container.rs

```diff
@@ -47,36 +60,76 @@ impl DOMContainerData {
 /// A collection for holding 3D `Object`s.
 #[derive(Debug)]
 pub struct DOMContainer {
-    pub dom          : HtmlElement,
-    resize_observer  : Option<ResizeObserver>,
-    data             : Rc<RefCell<DOMContainerData>>,
+    pub dom               : HtmlElement,
+    resize_observer       : Option<ResizeObserver>,
+    intersection_observer : Option<IntersectionObserver>,
+    data                  : Rc<RefCell<DOMContainerData>>,
 }
 
-impl DOMContainer {
-    pub fn new(dom_id:&str) -> Result<Self> {
-        let dom : HtmlElement = dyn_into(get_element_by_id(dom_id)?)?;
+impl Clone for DOMContainer {
+    fn clone(&self) -> Self {
+        DOMContainer::from_element(self.dom.clone())
+    }
+}
 
-        let width           = dom.client_width()  as f32;
-        let height          = dom.client_height() as f32;
-        let dimensions      = Vector2::new(width, height);
-        let data            = Rc::new(RefCell::new(DOMContainerData::new(dimensions)));
-        let resize_observer = None;
-        let mut ret         = Self { dom, resize_observer, data };
+impl DOMContainer {
+    pub fn from_element(dom:HtmlElement) -> Self {
+        let rect                  = dom.get_bounding_client_rect();
+        let x                     = rect.x()      as f32;
+        let y                     = rect.y()      as f32;
+        let width                 = rect.width()  as f32;
+        let height                = rect.height() as f32;
+        let dimensions            = Vector2::new(width, height);
+        let position              = Vector2::new(x    , y);
+        let data                  = DOMContainerData::new(position, dimensions);
```

Comment:
This should be refactored to a funciton. It is just translation of `get_bounding_client_rect` to internal representaiton.

Reply:
@wdanilo sorry, what's the suggestion here?